HeroVault/HeroVault/Features/Game/GameScreenViewController.swift

```diff
@@ -0,0 +1,179 @@
+        handleIncorrectAnswer(for: currentCell, collectionView: collectionView)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView,
+                        layout collectionViewLayout: UICollectionViewLayout,
+                        sizeForItemAt indexPath: IndexPath) -> CGSize {
+        CGSize(width: 170, height: 210)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let gameScreenCollectionViewCell =
+                gameCollectionView.dequeueReusableCell(withReuseIdentifier:
+                                                        Constants.SegueIdentifierNames.gameScreenCollectionViewCellName,
+                                                       for: indexPath) as? GameScreenCollectionViewCell else {
+            return UICollectionViewCell()
+        }
+        gameScreenCollectionViewCell.setUpNib(gameImage: gameScreenViewModel.populateCellImage)
+        return gameScreenCollectionViewCell
+    }
+    
+    private func handleCorrectAnswer(for cell: GameScreenCollectionViewCell) {
+        cell.correctAnswer()
+        reloadView()
+        currentScoreNumber()
+    }
+    
+    private func handleIncorrectAnswer(for cell: GameScreenCollectionViewCell, collectionView: UICollectionView) {
+        cell.incorrectAnswer()
+        gameScreenViewModel.cellsClickable = false
+        gameScreenViewModel.newHighScore() ? showNewHighScore() : showError()
+        gameScreenViewModel.resetScore()
+        highlightCorrectCell(in: collectionView)
+    }
+    
+    private func highlightCorrectCell(in collectionView: UICollectionView) {
+        guard let correctIndex = gameScreenViewModel.correctIndex else {
+            return
+        }
+        guard let correctCell = collectionView.cellForItem(at: IndexPath(row: correctIndex, section: 0)) as?
+                GameScreenCollectionViewCell else {
+            return
+        }
+        correctCell.correctAnswer()
+    }
+}
+
+extension GameScreenViewController: ViewModelProtocol {
+    func startLoadingIndicator() {
+        view.showLoadingIndicator()
+        gameCollectionView.isHidden = true
+        currentScoreLabel.isHidden = true
+        playButtonLabel.isHidden = true
+        highScoreLabel.isHidden = true
+        instructionLabel.isHidden = true
+    }
+    
+    func stopLoadingIndicator() {
+        view.stopLoadingIndicator()
+        gameCollectionView.isHidden = false
+        currentScoreLabel.isHidden = false
+        highScoreLabel.isHidden = false
+        instructionLabel.isHidden = false
+    }
+    
+    func reloadView() {
+        setUpView()
+        setUpCollectionView()
+        gameCollectionView.reloadData()
+    }
+    
+    func showError() {
+        let alert = UIAlertController(title: "Game Over", message:  "Your score was: \(gameScreenViewModel.score)", preferredStyle: .alert)
```

Comment:
Rather add this function to your UIView+extension class for reusbility
extension UIViewController {
    func showAlert(title: String, message: String, actions: @escaping () -> Void) {
        let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
        alert.addAction(UIAlertAction(title: "OK", style: .default) { _ in
            actions()
        })
        self.present(alert, animated: true, completion: nil)
    }
}

Reply:
Doing this will allow you to do more erro handling throughout the pp by using an alert